lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -68,209 +47,25 @@ def provider
-      def merge_response!(other)
-        @_response.concat(other.response)
-        self
-      end
-
-      private
-
-      sig { params(node: Prism::Node, name: String, command: String, kind: Symbol).void }
-      def add_test_code_lens(node, name:, command:, kind:)
-        # don't add code lenses if the test library is not supported or unknown
-        return unless SUPPORTED_TEST_LIBRARIES.include?(DependencyDetector.instance.detected_test_library) && @path
-
-        arguments = [
-          @path,
-          name,
-          command,
-          {
-            start_line: node.location.start_line - 1,
-            start_column: node.location.start_column,
-            end_line: node.location.end_line - 1,
-            end_column: node.location.end_column,
-          },
-        ]
-
-        grouping_data = { group_id: @group_id_stack.last, kind: kind }
-        grouping_data[:id] = @group_id if kind == :group
-
-        @_response << create_code_lens(
-          node,
-          title: "Run",
-          command_name: "rubyLsp.runTest",
-          arguments: arguments,
-          data: { type: "test", **grouping_data },
-        )
-
-        @_response << create_code_lens(
-          node,
-          title: "Run In Terminal",
-          command_name: "rubyLsp.runTestInTerminal",
-          arguments: arguments,
-          data: { type: "test_in_terminal", **grouping_data },
-        )
-
-        @_response << create_code_lens(
-          node,
-          title: "Debug",
-          command_name: "rubyLsp.debugTest",
-          arguments: arguments,
-          data: { type: "debug", **grouping_data },
-        )
-      end
-
-      sig { params(gem_name: Prism::StringNode).returns(T.nilable(String)) }
-      def resolve_gem_remote(gem_name)
-        spec = Gem::Specification.stubs.find { |gem| gem.name == gem_name.content }&.to_spec
-        return if spec.nil?
-
-        [spec.homepage, spec.metadata["source_code_uri"]].compact.find do |page|
-          page.start_with?("https://github.com", "https://gitlab.com")
-        end
-      end
-
-      sig { params(class_name: String, method_name: T.nilable(String)).returns(String) }
-      def generate_test_command(class_name:, method_name: nil)
-        command = BASE_COMMAND + T.must(@path)
-
-        case DependencyDetector.instance.detected_test_library
-        when "minitest"
-          command += if method_name
-            " --name " + "/#{Shellwords.escape(class_name + "#" + method_name)}/"
-          else
-            " --name " + "/#{Shellwords.escape(class_name)}/"
-          end
-        when "test-unit"
-          command += " --testcase " + "/#{Shellwords.escape(class_name)}/"
-
-          if method_name
-            command += " --name " + Shellwords.escape(method_name)
-          end
+      sig { override.returns(ResponseType) }
+      def response
+        result = []
+        @listeners.each do |listener|
+          result.concat(listener.response) if listener.response
```

Comment:
Do we need the nilable check here? We default to returning empty arrays for code lens listeners.